fix broken url comparision test with python3.8.8

the python3.8.8 release caused a regression in one of our tests:
```
_______________________________ test_urlparams ________________________________
[gw0] linux -- Python 3.8.8 /home/circleci/.pyenv/versions/3.8.8/bin/python3
src/olympia/amo/tests/test_helpers.py:201: in test_urlparams
    amo.tests.assert_url_equal(s, '%s?sort=name&amp;x=y#frag' % url)
        c          = {'base': '/en-US/firefox/themes/category',
 'base_frag': '/en-US/firefox/themes/category#hash',
 'base_query': '/en-US/firefox/themes/category?x=y',
 'frag': 'frag',
 'sort': 'name'}
        s          = '/en-US/firefox/themes/category?x=y&amp;sort=name#frag'
        url        = '/en-US/firefox/themes/category'
src/olympia/amo/tests/__init__.py:233: in assert_url_equal
    compare_url_part(parse_qs(parsed.query), parse_qs(parsed_expected.query))
        compare_host = False
        expected   = '/en-US/firefox/themes/category?sort=name&amp;x=y#frag'
        parsed     = ParseResult(scheme='', netloc='', path='/en-US/firefox/themes/category', params='', query='x=y&amp;sort=name', fragment='frag')
        parsed_expected = ParseResult(scheme='', netloc='', path='/en-US/firefox/themes/category', params='', query='sort=name&amp;x=y', fragment='frag')
        url        = '/en-US/firefox/themes/category?x=y&amp;sort=name#frag'
src/olympia/amo/tests/__init__.py:239: in compare_url_part
    assert part == expected, 'Expected %s, got %s' % (expected, part)
E   AssertionError: Expected {'sort': ['name'], 'amp;x': ['y']}, got {'x': ['y'], 'amp;sort': ['name']}
        expected   = {'amp;x': ['y'], 'sort': ['name']}
        part       = {'amp;sort': ['name'], 'x': ['y']}
```
related: https://bugs.python.org/issue42967

--- a/src/olympia/amo/templatetags/jinja_helpers.py
+++ b/src/olympia/amo/templatetags/jinja_helpers.py
@@ -30,11 +30,15 @@ register = Library()
 
 
 # Registering some utils as filters:
-urlparams = library.filter(utils.urlparams)
 library.filter(utils.epoch)
 library.filter(utils.isotime)
 library.global_function(dict)
 library.global_function(utils.randslice)
+
+
+@library.filter
+def urlparams(*args, **kwargs):
+    return jinja2.Markup(utils.urlparams(*args, **kwargs))
 
 
 @library.global_function

--- a/src/olympia/amo/tests/test_helpers.py
+++ b/src/olympia/amo/tests/test_helpers.py
@@ -244,12 +244,21 @@ def test_urlparams_returns_safe_string():
         '{{ unsafe_url|urlparams }}',
         {
             'unsafe_url': "http://url.with?foo=<script>alert('awesome')</script>"
-            "&baa=that"
+            '&baa=that'
         },
     )
     assert string == (
         'http://url.with?foo=%3Cscript%3Ealert%28%27awesome%27%29%3C%2Fscript%3E'
         '&baa=that'
+    )
+
+    string = render(
+        '{{ "http://safe.url?baa=that"|urlparams(foo=unsafe_param) }}',
+        {'unsafe_param': "<script>alert('awesome')</script>"},
+    )
+    assert string == (
+        'http://safe.url?baa=that'
+        '&foo=%3Cscript%3Ealert%28%27awesome%27%29%3C%2Fscript%3E'
     )
 
 

--- a/src/olympia/amo/tests/test_helpers.py
+++ b/src/olympia/amo/tests/test_helpers.py
@@ -198,7 +198,7 @@ def test_urlparams():
 
     # Adding query with existing params.
     s = render('{{ base_query|urlparams(frag, sort=sort) }}', c)
-    amo.tests.assert_url_equal(s, '%s?sort=name&amp;x=y#frag' % url)
+    amo.tests.assert_url_equal(s, '%s?sort=name&x=y#frag' % url)
 
     # Replacing a query param.
     s = render('{{ base_query|urlparams(frag, x="z") }}', c)
@@ -239,6 +239,18 @@ def test_urlparams_returns_safe_string():
 
     s = render('{{ "https://foo.com/"|urlparams(param="%AAA") }}', {})
     assert s == 'https://foo.com/?param=%AAA'
+
+    string = render(
+        '{{ unsafe_url|urlparams }}',
+        {
+            'unsafe_url': "http://url.with?foo=<script>alert('awesome')</script>"
+            "&baa=that"
+        },
+    )
+    assert string == (
+        'http://url.with?foo=%3Cscript%3Ealert%28%27awesome%27%29%3C%2Fscript%3E'
+        '&baa=that'
+    )
 
 
 def test_isotime():

--- a/src/olympia/amo/utils.py
+++ b/src/olympia/amo/utils.py
@@ -123,18 +123,17 @@ def urlparams(url_, hash=None, **query):
     fragment = hash if hash is not None else url.fragment
 
     # Use dict(parse_qsl) so we don't get lists of values.
-    q = url.query
-    query_dict = dict(parse_qsl(force_str(q))) if q else {}
+    query_dict = dict(parse_qsl(force_str(url.query))) if url.query else {}
     query_dict.update(
         (k, force_bytes(v) if v is not None else v) for k, v in query.items()
     )
     query_string = urlencode(
         [(k, unquote_to_bytes(v)) for k, v in query_dict.items() if v is not None]
     )
-    new = ParseResult(
+    result = ParseResult(
         url.scheme, url.netloc, url.path, url.params, query_string, fragment
     )
-    return new.geturl()
+    return result.geturl()
 
 
 def partial(func, *args, **kw):

